docs/hybrid-slurm-cluster/inter-gcp-project-hybrid-slurm.md

```diff
@@ -0,0 +1,536 @@
+
+## Instructions
+
+### Before you begin
+* Build ghpc
+
+#### Select or Create 2 GCP Projects
+
+This process will require 2 projects:
+
+* Project A: Where the simulated “On-premise” static slurm cluster will be
+  deployed.
+* Project B: Where the cloud partitions will create new compute VM instances to
+  complete slurm jobs.
+
+Identify the 2 projects you intend to use. "Project A" and "Project B" will be
+referred to in the rest of this document based on these definitions.
+
+#### Enable Required APIs
+
+The following APIs are required to complete this demo:
+
+* [Compute Engine API][computeapi]
+* [Cloud DNS API][clouddnsapi]
+
+[computeapi]: https://cloud.google.com/compute/docs/reference/rest/v1
+[clouddnsapi]: https://cloud.google.com/dns/docs/reference/v1
+
+#### Set IAM Roles
+The service account attaches to the slurm controller in Project A
+([see above](#select-or-create-2-gcp-projects))
+must have the Editor role in
+Project A and Project B. If not specified, this will be the
+[default compute engine service account][computesa].
+
+[computesa]:https://cloud.google.com/compute/docs/access/service-accounts#default_service_account
+
+#### Dependencies
+This demo has the same baseline dependencies as the HPC Toolkit that are
+outlined in the main [README.md](../../README.md#dependencies).
+
+In addition, some pip packages need to be installed locally. Run the following
+command to install the pip packages outlined in
+[requirements.txt](./requirements.txt):
+
+```shell
+pip install -r requirements.txt
+```
+
+#### Build ghpc
+
+Before you begin, ensure that you have built the `ghpc` tool in the HPC Toolkit.
+For more information see the [README.md](../../README.md#quickstart) Quickstart.
+
+### Create VPC Networks
+A blueprint for creating VPC networks in each project that can support network
+and DNS peering can be found at [create-networks.yaml]. This
+blueprint will do the following:
+
+* Create a network named `static-cluster-network` in project A.
+* Create a subnetwork of `static-cluster-network` named `primary-subnet` with
+  an internal IP range of 10.0.0.0/16.
+* Create a network named `compute-vpc-network` in project B.
+* Create a subnetwork of `compute-vpc-network` named `primary-subnet` with an
+  internal IP range of 10.1.0.0/16
+
+Create a deployment directory for the networks using `ghpc`:
+
+```shell
+ghpc create blueprints/create-networks.yaml --vars project_id="<<Project_A_ID>>",project_id_compute="<<Project_B_ID>>"
+```
+
+If successful, this command will provide 3 terraform operations that can be
+performed to deploy the deployment directory. They should look similar to the
+following:
+
+```shell
+Terraform group was successfully created in directory peering-networks-demo/primary
+To deploy, run the following commands:
+  terraform -chdir=peering-networks-demo/primary init
+  terraform -chdir=peering-networks-demo/primary validate
+  terraform -chdir=peering-networks-demo/primary apply
```

Comment:
Experienced error on apply:
```
│ Error: Invalid function argument
│ 
│   on modules/vpc/main.tf line 21, in locals:
│   21:   cidr_blocks                 = cidrsubnets(var.network_address_range, local.primary_subnetwork_new_bits, var.additional_subnetworks[*].new_bits...)
│     ├────────────────
│     │ local.primary_subnetwork_new_bits is 0
│ 
│ Invalid value for "newbits" parameter: must extend prefix by at least one bit.
```

from offline discussion, this is going to be resolved by an update @heyealex has in work for this blueprint. Reassign back to me when it is ready for another look.

Reply:
This should be resolved in the recent commit.